### DISCUSSION TITLE
Multiple Results for City

### DISCUSSION THREAD

**Discussion by pmad99** [JaouherK/streamDeck-weatherPlugin#115] [General]
When i look for my city using Open Weather, i get multiple results, 5 to be exact. Mine is the fourth, so when i put this into the city name field on my Stream Deck software, i end up getting temps for the first one, and some other one i have no idea where its coming from?

How can i get this to specifically show just my city only? 

I did do a search, then clicked on the 4th link which is my city, and copied the # from the link: https://openweathermap.org/city/#######

I used that, it didnt work either.

**Comment by brstegge** [comment]
I have this same issue. Open Weather returns 3 results in the US for my city name, there is no way to specify which one to use, and the button throws an error as a result. I tried using the city UID that @pmad99 mentioned above, and using the geo coordinates from Open Weather, but neither worked.

**Reply by ragemachinest** [reply]
@brstegge I don't know that workarounds are ideal around here but I just added this to Issue #160

I got the same thing and don't know if it was a temporary issue or what. If I knew how to code I'd fix it but I did find a workaround, at least for me using OpenWeather. First fill out the details in Stream Deck with the API Key, City Name, Temp, etc. Go find the Coordinates on OpenWeather for your city.

Edit this file (Windows): C:\Users\YourWindowsUser\AppData\Roaming\Elgato\StreamDeck\Plugins\com.jk.weather.sdPlugin\plugin\main.js

Line 98, replace the URL with this, substituting your city's coordinates (lat, lon) from OpenWeather. Save the file. Restart Stream Deck:

[https://api.openweathermap.org/data/2.5/weather?lat=32.7668&lon=-96.7836&appid=${apiKey}&units=${u}](https://api.openweathermap.org/data/2.5/weather?lat=32.7668&lon=-96.7836&appid=$%7BapiKey%7D&units=$%7Bu%7D)
